src/pkcs15init/pkcs15-myeid.c

```diff
@@ -510,10 +510,10 @@ myeid_store_key(struct sc_profile *profile, struct sc_pkcs15_card *p15card,
 				keybits = key_info->field_length;
 			else
 				key_info->field_length = keybits;
-			break;
-			
+
 			if (sc_card_find_ec_alg(p15card->card, keybits, &prkey->u.ec.params.id) == NULL)
 				LOG_TEST_RET(ctx, SC_ERROR_INVALID_ARGUMENTS, "Unsupported algorithm or key size");			
+			break;
```

Comment:
The `LOG_TEST_RET(..., SC_ERROR_INVALID_ARGUMENTS, ...)` macro always returns `SC_ERROR_INVALID_ARGUMENTS`, so `break;` is redundant (does not do anything) here.  The code would be more clear if `SC_FUNC_RETURN()` was used instead of `LOG_TEST_RET()`.